crates/hotshot-state-prover/src/circuit.rs

```diff
@@ -304,9 +391,17 @@ mod tests {
             fee_ledger_comm: F::default(),
             stake_table_comm: (F::default(), F::default(), F::default()),
         };
+        let bad_state_msg = [F::default(); 7];
```

Comment:
Maybe we can set everything correct except for the bls key commitment.

Reply:
done in 7669200

Reply:
Wait you cannot do this.
The stake table commitment check ignores the bls key part. If everyone sign this, the validation shall pass.

Reply:
But the signature verification circuit is checked over the message that contains the bls key commitment in the public input, the validation should fail in this case.

Reply:
a better test is provided in 6ff6fcf